pygui/__init__.py

```diff
@@ -212,6 +213,18 @@ def reconcile_children(self, wip_fiber: Fiber, elements: List[VNode]):
                 new_fiber = Fiber(
                     type=old_fiber.type,
                     props=element.props,
+                    # Have to create a snapshot of the state somewhere to
+                    # be able to reconcile properly... If the state has been
+                    # changed directly, there is no way of knowing what the
+                    # previous state looked like...
+                    # Another option is to let the renderer handle if there
+                    # should be any changes made to the dom object in the
+                    # `set_attribute` method :/ and thus leave out the comparison
+                    # between old and new props. That would also require to
+                    # combine the `set_attribute` and `clear_attribute`...
+                    # Vue's render API only has `patchProp`, but it still takes
+                    # a `prevValue` and a `nextValue`...
+                    raw_props=to_raw(element.props),
```

Comment:
Maybe I am wrong, but I always thought that components (and their children) are rerendered because their _state_ has changed, and that means they may (or may not) pass new props down. The top level component's props never change, too. Doesn't that "fix" this problem?